lib/dgram.js

```diff
@@ -666,6 +666,13 @@ function doSend(ex, self, ip, list, address, port, callback) {
   else
     err = state.handle.send(req, list, list.length, !!callback);
 
+  if (err >= 1) {
+    // Synchronous finish.
+    if (callback)
+      process.nextTick(callback, null, err - 1);
```

Comment:
Can you document here that the C++ code returns `msg_size + 1` to let the caller distinguish between sync and async return values?